app/components/post_draft/draft_input/index.tsx

```diff
@@ -133,16 +133,102 @@ export default function DraftInput({
     const intl = useIntl();
     const serverUrl = useServerUrl();
     const theme = useTheme();
+    const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false);
+    const [isEmojiPickerFocused, setIsEmojiPickerFocused] = useState(false);
 
     const handleLayout = useCallback((e: LayoutChangeEvent) => {
         updatePostInputTop(e.nativeEvent.layout.height);
     }, []);
 
+    // To keep track of cursor position and to retrieve the latest value of cursor position
+    const cursorPositionRef = useRef(cursorPosition);
+    useEffect(() => {
+        cursorPositionRef.current = cursorPosition;
+    }, [cursorPosition]);
```

Comment:
Why using a ref and not using the prop we already have? I could understand we use the ref if we expect asynchronous actions happening, but I don't see anything as such. I may be missing something.

Another reason could be to avoid re-creating some callbacks to avoid re-renders, but not sure if that is something we are looking into here.

Reply:
I switched to using `useRef` for `cursorPosition` to make sure we always have the latest value inside the `updateValue` function. React's state can sometimes get stuck with an outdated value in closures because they capture the state at the time they're created and that’s the reason why I was always getting cursorPosition value as 0. By using useRef, we avoid this problem since the .current property always holds the most up-to-date value without triggering re-renders. This ensures that cursorPosition is accurate when updating.

Reply:
That is true and makes sense if the callback is going to be called asynchronously. So... Something like this can happen:
- I set the state to value 1
- I create the callback with value 1
- I call the callback with value 1, and the callback is asynchronous
- While the callback is waiting, I set the state to 2
- The callback keeps executing with value 1 instead of value 2

But not sure if that is the scenario here. Here the following should be happening.
- I set the state to value 1
- I create the callback with value 1
- I set the state to value 2, so the callback is recreated before it can be called
- I use the callback with value 2 already

I may be missing the asynchronous nature of the callback where it is used, but if it is always synchronous, this may be hiding other problems (like bad memoization or missing dependencies down the line).

Reply:
As discussed over the call, the component was re-rendering on the dependencies of the useCallback was updated. Need a feature investigation on this.